std/xml.d

```diff
@@ -562,7 +562,7 @@ class Document : Element
         string tagString = xml.tag.tagString;
 
         this(xml.tag);
-        prolog = s[0 .. tagString.ptr - s.ptr];
+        prolog = s[0 .. &tagString[0] - &s[0]];
```

Comment:
This is one of the new safety checks that Walter added. What if `tagString` is empty? `.ptr` points past the end of the array. This way bounds checking is always done.